doc/arch/014-reporting-etl-pipeline.md

```diff
@@ -0,0 +1,150 @@
+* As of Jan 2020, what drugs were prescribed to Ashish?
+* As of Feb 2020, what drugs were prescribed to Ashish?
+* As of Mar 2020, what drugs were prescribed to Ashish?
+* ...and so on
+
+Since these tables are pre-computing questions like "What was the most recent BP as of X month", the reports queries
+become much simpler. Instead of dealing with this question, the reports query can instead fetch all records for a
+particular month, and the "most recent" information will be readily available in the record.
+
+### Calendar Table
+
+Time, down to the month, is an important dimension of all reporting tables. As such, a calendar table will be created to
+support a consistent and robust way to describe months across all reporting tables. Calendar tables are a common
+occurrence in data warehouse design ([Here's one such example](https://www.mssqltips.com/sqlservertip/4054/creating-a-date-dimension-or-calendar-table-in-sql-server/)).
+
+The important elements of our calendar will be:
+* Cardinal month (1-12)
+* Year
+* Quarter: Quarterly analysis is common, so precomputing will be beneficial.
+* A unique date, like first of the month: This helps uniquely identify or group by each record in the calendar table on
+  a single column. `GROUP BY (year, month)` is more awkward.
+
+The calendar table will usually be joined with Simple's application tables to populate the denormalized reporting
+tables. The calendar table will be implemented as a non-materialized database view, since it is fairly small and not
+expensive to compute when needed.
+
+### Reports
+
+Since reporting queries will be simplified, we can start using Metabase for new reports. Metabase is ideal for quick
+prototyping and validation of reports with stakeholders because:
+
+* Easy-to-use interface
+* Easy to grant access to stakeholders, and controlled access to ourselves if necessary
+* Dynamic queries (eg. pick your own date range)
+
+Once reports are validated, we may want to move them in-house into the Simple Server application/dashboard. The benefits
+of doing so would be:
+
+* Better test coverage
+* Less fragmentation
+
+In any case, the long-term location of the reports themselves will not greatly impact the reporting table design
+described here.
+
+## Alternatives and constraints
+
+Apart from the proposed approach of reporting tables as database views, there are some alternatives that were
+considered.
+
+### Reporting tables in a separate database
+
+We could continue to generate the reporting tables by hand, but place them in a separate database. This would greatly
+alleviate performance problems. However, database performance is a secondary concern. Furthermore, the proposed approach
+will still set us up nicely to extract the reporting tables to a separate database in the future if necessary.
+
+### Rollup summary tables
+
+Rollup summary tables were proposed in [an earlier ADR](011-rollup-summary-tables-for-dashboards.md). The crux of the
```

Comment:
I think marking that ADR as rejected would be fine here, or in whatever PR marks this as accepted.